dspace-server-webapp/src/test/java/org/dspace/app/rest/BitstreamRestControllerIT.java

```diff
@@ -396,6 +562,141 @@ public void testPrivateBitstream() throws Exception {
+                                .andExpect(status().isOk());
+
+            checkNumberOfStatsRecords(bitstream, 1);
+
+            String tokenEPerson2 = getAuthToken(eperson2.getEmail(), "qwerty02");
+            getClient(tokenEPerson2).perform(get("/api/core/bitstreams/" + bitstream.getID() + "/content"))
+                                .andExpect(status().isForbidden());
+
+            // Anonymous users CANNOT access/download Bitstreams that are restricted
+            getClient().perform(get("/api/core/bitstreams/" + bitstream.getID() + "/content"))
+                                .andExpect(status().isUnauthorized());
+
+            checkNumberOfStatsRecords(bitstream, 1);
+        }
+    }
+
+    @Test
+    public void restrictedGroupBitstreamAccessGrantByAdminsTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson adminParentCommunity = EPersonBuilder.createEPerson(context)
+                .withEmail("adminCommunity@mail.com")
+                .withPassword("qwerty00")
+                .build();
+
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                .withName("Parent Community")
+                .withAdminGroup(adminParentCommunity)
+                .build();
+
+        EPerson adminCol1 = EPersonBuilder.createEPerson(context)
+                .withEmail("admin1@mail.com")
+                .withPassword("qwerty01")
+                .build();
+        Collection col1 = CollectionBuilder.createCollection(context, parentCommunity)
+                .withName("Collection 1")
+                .withAdminGroup(adminCol1)
+                .build();
+
+        EPerson adminCol2 = EPersonBuilder.createEPerson(context)
+                .withEmail("admin2@mail.com")
+                .withPassword("qwerty02")
+                .build();
+        Collection col2 = CollectionBuilder.createCollection(context, parentCommunity)
+                .withName("Collection 2")
+                .withAdminGroup(adminCol2)
+                .build();
+
+        Group restrictedGroup = GroupBuilder.createGroup(context)
+                .withName("Restricted Group")
+                .addMember(eperson)
+                .build();
+
+        String bitstreamContent = "Private!";
+        try (InputStream is = IOUtils.toInputStream(bitstreamContent, CharEncoding.UTF_8)) {
+
+            Item item = ItemBuilder.createItem(context, col1)
+                    .withTitle("item")
+                    .withIssueDate("2018-10-17")
+                    .withAuthor("Doe, John")
+                    .build();
+            Bitstream bitstream = BitstreamBuilder
+                .createBitstream(context, item, is)
+                .withName("Test Embargoed Bitstream")
+                .withDescription("This bitstream is embargoed")
+                .withMimeType("text/plain")
+                .withReaderGroup(restrictedGroup)
+                .build();
```

Comment:
Again, I think you can close the try clause here. Also, you are missing a context.restoreAuthSystemState(); in this test too.